pkg/app/mutate_app_pss_compliance.go

```diff
@@ -0,0 +1,142 @@
+package app
+
+import (
+	"context"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	"github.com/giantswarm/app/v7/pkg/key"
+	"github.com/giantswarm/k8smetadata/pkg/label"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/app-admission-controller/pkg/mutator"
+	"github.com/giantswarm/app-admission-controller/pkg/project"
+)
+
+var (
+	// pssCutoffVersion represents first & lowest Giant Swarm Release version
+	// which does not support PodSecurityPolicies.
+	pssCutoffVersion, _ = semver.NewVersion("v19.2.0")
+)
+
+const (
+	extraConfigName   = "pss-compliance-patch"
+	extraConfigValues = `global:
+  podsecuritystandards:
+    enforced: true`
+	topPriority = 150
+)
+
+// mutateConfigForPSSCompliance is a temporary solution to
+// https://github.com/giantswarm/roadmap/issues/2716. Revert once migration to
+// Release >= v19.2.0 is complete and managed apps no longer rely on PSPs.
+func (m *Mutator) mutateConfigForPSSCompliance(ctx context.Context, app v1alpha1.App) ([]mutator.PatchOperation, error) {
+	result := []mutator.PatchOperation{}
+	clusterID := key.ClusterLabel(app)
+	if clusterID == "" {
+		// This App CR does not belong to any Workload Cluster - it does not need any more patches.
+		return result, nil
+	}
+
+	extraConfig := v1alpha1.AppExtraConfig{
+		Kind:      "configMap",
+		Name:      extraConfigName,
+		Namespace: app.Namespace,
+		Priority:  topPriority,
+	}
+
+	// If extraConfigs are already patched with 'extraConfigName', let's save
+	// ourselves some checks, ensure ConfigMap, and assume everything is in
+	// order.
+	ec := key.ExtraConfigs(app)
+	if len(ec) > 0 && ec[len(ec)-1] == extraConfig {
+		if err := m.ensureConfigMap(ctx, app.Namespace); err != nil {
+			return nil, microerror.Mask(err)
+		}
+		return result, nil
+	}
+
+	// This App belongs to a Workload Cluster, which is using a certain Release
+	// version. Let's determine what it is.
+	var releaseVersion *semver.Version
+	{
+		// We don't want to guess Cluster's namespace because it's been
+		// historically difficult. Cluster ID/name is unique, so we are relying
+		// on that.
+		clusterCRList := capiv1beta1.ClusterList{}
+		err := m.k8sClient.CtrlClient().List(ctx, &clusterCRList, &client.ListOptions{})
```

Comment:
This is currently listing all Cluster CRs, not using the cluster ID for the App CR

Reply:
Second idea: maybe a test where multiple cluster CRs are present in different versions is useful here

Reply:
Yeah, good catch. I have replaced a `Get` with `List` to get rid of namespacing issues, but never updated the code accordingly.
Patched, test on the way.